drivers/crypto/crypto_stm32.c

```diff
@@ -21,12 +19,34 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(crypto_stm32);
 
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_cryp)
+#define DT_DRV_COMPAT st_stm32_cryp
+#elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32_aes)
+#define DT_DRV_COMPAT st_stm32_aes
```

Comment:
Can both be enabled at the same time on the same SoC? If so, can you please make the driver multi-instance and support both cases?

Note you can do something like this to handle that within a single driver if that makes sense:

```c
#define DT_DRV_COMPAT st_stm32_cryp
DT_INST_FOREACH_STATUS_OKAY(DEFINE_CRYP)
#undef DT_DRV_COMPAT
#define DT_DRV_COMPAT st_stm32_aes
DT_INST_FOREACH_STATUS_OKAY(DEFINE_AES)
```

If the device is a singleton and no SoC contains both, then this PR LGTM too.

Reply:
Thanks for this suggestions!
In this case no SoC contains both, but only a single instance of either AES or CRYP.